xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/doc/XWikiDocumentMockitoTest.java

```diff
@@ -1397,4 +1399,37 @@ public void tofromXMLDocument() throws XWikiException
 
         assertEquals(this.document, newDocument);
     }
+    
+    @Test
+    public void listAddAttachment() throws XWikiException{
+        XWikiAttachment attachment = new XWikiAttachment(this.document, "testAttachment");
+        attachmentList.add(attachment);
+        assertEquals(this.document.getAttachmentList().contains(attachment),true);
+    }
+    
+    @Test
+    public void listMaintainsOrder() throws XWikiException{
+        XWikiAttachment attachment1 = new XWikiAttachment(this.document, "attachmentA");
+        XWikiAttachment attachment2 = new XWikiAttachment(this.document, "attachmentB");
+        XWikiAttachment attachment3 = new XWikiAttachment(this.document, "attachmentC");
+        attachmentList.add(attachment3);
+        attachmentList.add(attachment1);
+        attachmentList.add(attachment2);
+        List<XWikiAttachment> list = new ArrayList<XWikiAttachment>();
+        list.add(attachment1);
+        list.add(attachment2);
+        list.add(attachment3);
+        assertEquals(this.document.getAttachmentList(),list);
+    }
+    
+    @Test
+    public void testTest() throws XWikiException{
+        assertEquals("hi","hello");
+    }
+    
+    @Test
+    public void listClear() throws XWikiException{
+        attachmentList.clear();
+        assertEquals(attachmentList,new ArrayList<XWikiAttachment>());
```

Comment:
I would assertTrue attachmentList.isEmpty() instead since that's what you want to know (not that it's equals to some list).